### DISCUSSION TITLE
hx-post triggered on enter key in <textarea>

### DISCUSSION THREAD

**Discussion by jritsema** [bigskysoftware/htmx#2467] [Q&A] [answered]
I'm currently triggering an `hx-post` from a button click, however, I want to trigger it when the user hits the enter key inside the `<textarea>`.  What is the simplest way to do this in htmx?

```html
<textarea
  name="question"
  class="form-control"
  rows="4"
></textarea>

<button
  id="ask"
  type="button"
  hx-post="/ask"
  hx-target="#content"
  hx-indicator="#indicator"
>
  Ask
</button>
```

**Comment by jritsema** [comment]
I figured it out but I'm not yet preventing the new line from showing up in the <textarea>.

```html
<textarea
  name="question"
  hx-trigger="keydown[key==='Enter'&&!shiftKey]"
  hx-post="/ask"
  hx-target="#content"
  hx-indicator="#indicator"
></textarea>
```

**Reply by andryyy** [reply, marked as answer]
Hey, you can add `hx-on:keydown="(event.keyCode===13&&!event.shiftKey)?event.preventDefault():null"`. This does allow for Shift + Enter for new lines but send on Enter.

**Reply by jritsema** [reply]
Thank you @andryyy!
